torch/csrc/jit/codegen/cuda/kernel.h

```diff
@@ -15,19 +17,32 @@ namespace fuser {
 namespace cuda {
 namespace kir {
 
+class ExprBuilderVisitor;
+
 //! Summary of interesting facts about the kernel
 struct KernelSummary {
+  std::unique_ptr<glfdc::ExprDAG> memoized_dag;
```

Comment:
Please comment the new fields.

Reply:
Where is `memoized_dag` used?

Reply:
Every memoized expression (glfdc::Expr and later glfdc::ExprEvaluator) refers to the part of DAG (but not owns) , thus we have to ensure it is alive.
Reference counting schemes in that case would be overkill IMO.

Reply:
Added comments

Reply:
Thanks. Pease add a comment to `memoized_dag` as well.